app/src/main/java/com/asfoundation/wallet/billing/paypal/PayPalIABViewModel.kt

```diff
@@ -170,6 +181,46 @@ class PayPalIABViewModel @Inject constructor(
     )
   }
 
+  fun openUrlCustomTab(context: Context, url: String) {
+    if (runningCustomTab) return
+    runningCustomTab = true
+    val customTabsBuilder = CustomTabsIntent.Builder().build()
+    customTabsBuilder.intent.setPackage(GooglePayWebFragment.CHROME_PACKAGE_NAME)
+    customTabsBuilder.launchUrl(context, Uri.parse(url))
+  }
+
+  fun processPayPalResult(amount: BigDecimal, currency: String, transactionBuilder: TransactionBuilder, origin: String?) {
+    if (isFirstResultRun) {
```

Comment:
this seems a strange behavior. maybe the problem lies on another place for adding this logic...